tanner/sessions/session_analyzer.py

```diff
@@ -27,18 +33,132 @@ async def analyze(self, session_key, redis_client):
 
-    async def save_session(self, redis_client):
+    async def save_session(self, redis_client, pg_client):
         while not self.queue.empty():
+            columns = (
+                'id, sensor_id, ip, port, {}, user_agent, start_time, '
+                'end_time, rps, atbr, accepted_paths, errors, hidden_links, referer'
+            )
+
             session = await self.queue.get()
-            s_key = session['snare_uuid']
-            del_key = session['sess_uuid']
+            s_key = session["snare_uuid"]
+            del_key = session["sess_uuid"]
+            print("Printing sessions")
+
+            start_time = datetime.fromtimestamp(session["start_time"]).strftime('%Y-%m-%d %H:%M:%S')
+            end_time = datetime.fromtimestamp(session["end_time"]).strftime('%Y-%m-%d %H:%M:%S')
+
+            # Some of the sessions have location as NA
+            try:
+                country_info = True
+                country = session["location"]["country"]
+                country_code = session["location"]["country_code"]
+                city = session["location"]["city"]
+                zip_code = session["location"]["zip_code"]
+            except (TypeError, KeyError):
+                country_info = False
+                location = session["location"]
+
             try:
-                await redis_client.zadd(s_key, session['start_time'], json.dumps(session))
+                # TODO: pg_client to insert
+                if country_info:
+                    all_columns = columns.format(
+                        "country, country_code, city, zip_code"
+                    )
+
+                    sessions_query = (
+                        "INSERT INTO sessions ({cols}) "
+                        "VALUES ('{uuid}','{sensor}','{ip}',{port},{country},"
+                        "{ccode},{city},{zcode},'{ua}','{st}','{et}',{rps},"
+                        "{atbr},{apaths},{err},{hlinks},'{referer}');"
+                    ).format(
+                        cols=all_columns,
+                        uuid=session["sess_uuid"],
+                        sensor=session["snare_uuid"],
+                        ip=session["peer_ip"],
+                        port=session["peer_port"],
+                        country=country,
+                        ccode=country_code,
+                        city=city,
+                        zcode=zip_code,
+                        ua=session["user_agent"],
+                        st=start_time,
+                        et=end_time,
+                        rps=session["requests_in_second"],
+                        atbr=session["approx_time_between_requests"],
+                        apaths=session["accepted_paths"],
+                        err=session["errors"],
+                        hlinks=session["hidden_links"],
+                        referer=session["referer"]
+                    )
+                else:
+                    all_columns = columns.format("country")
```

Comment:
I don't like such code duplication. Try to make only one request and put the data into unified format, you can use `get` on location and set default value if some data is not available